<table align="center" width=100%>
    <tr>
        <td width="15%">
            <img src="scala_takehome.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> Take-Home
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

**Import the required libraries**

In [1]:
# type your code here

import numpy as np
import pandas as pd

### Let's begin with some hands-on practice exercises

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. The experiment shows that the average monthly electricity consumption for a house is 458 units. The study records a total of 3206 units of electricity consumption. Calculate the duration of the experiment. Also, identify the criteria to calculate the median consumption. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [2]:
# type your code here

avg_consumption = 458
total_consumption = 3206

duration = total_consumption / avg_consumption

print('Duration: {} months.'.format(duration))

Duration: 7.0 months.


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b> 2. Find the wage above which 65% of wages would lie.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider the wages (in dollars) of 15 workers as given below:

            wage = [247, 228, 352, 298, 345, 312, 300, 285, 246, 352, 297, 304, 265, 325, 276]

In [3]:
# type your code here

wage = [247, 228, 352, 298, 345, 312, 300, 285, 246, 352, 297, 304, 265, 325, 276]
wage = pd.Series(wage)
print(wage.quantile(0.35))

284.1


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. Jimmy requires 38 minutes on average, with a standard deviation of 5 minutes to distribute the newspapers. On the other hand, Richard requires 27 minutes on average, with a standard deviation of 6 minutes to distribute the newspapers. Identify most consistent distributor.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [4]:
# type your code here

avg_J = 38
std_J = 5

avg_R = 27
std_R = 6

jimmy_coefficient_variation = (std_J/avg_J) * 100
richard_coefficient_variation = (std_R/avg_R) * 100
 
print('Jimmy:', jimmy_coefficient_variation)
print('Richard:', richard_coefficient_variation)

Jimmy: 13.157894736842104
Richard: 22.22222222222222


Jimmy is a consistent distributor as the coefficient of variation of Jimmy is less than that of Richard.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. Identify the extreme weights from the given list of weights using the interquartile range. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider the weights (in kg) of 10 students as given below:

            weight = [14.7, 15.2, 39.8, 44.5, 21.2, 28.5, 35.2, 19.7, 40.4, 16.5]

In [5]:
# type your code here

weight = [14.7, 15.2, 39.8, 44.5, 21.2, 28.5, 35.2, 19.7, 40.4, 16.5]
weight = pd.Series(weight)

Q1 = weight.quantile(0.25)
Q3 = weight.quantile(0.75)
IQR = Q3 - Q1

outliers = [x for x in weight if (x < Q1 - 1.5 * IQR) or (x > Q3 + 1.5 * IQR)]
outliers

[]

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. A car showroom in Ohio exhibits a total of 12 cars of different brands. Identify the type of skewness in the showroom prices of cars. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider the prices (in dollars) for 12 cars as given below:

            price = [14751, 16422, 15398, 9445, 12589, 11687, 10692, 8475, 11184, 9961, 12898, 11905]

In [6]:
# type your code here

price = [14751, 16422, 15398, 9445, 12589, 11687, 10692, 8475, 11184, 9961, 12898, 11905]
price = pd.Series(price)

mean = np.mean(price)
median = np.median(price)
std = np.std(price)

skewness = (3.0*(mean - median))/std
print('Skewness is:', skewness)

Skewness is: 0.41329390884304773


The skewness in the showroom prices of cars is positively skewed as the value of skewness is positive.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. A company offers its customers to take a vehicle and an eduacational loan. A study shows that the probability that a customer takes a vehicle loan is 0.35, the probability that a customer takes an educational loan is 0.46 and the probability that a customer takes at least one of the loan is 0.59. Find the probability that a customer takes both vehicle and educational loans.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [7]:
# type your code here

prob_V = 0.35
prob_E = 0.46
prob_V_or_E = 0.59

prob_V_and_E = (prob_V + prob_E) - prob_V_or_E
print('Probability is {}'.format(prob_V_and_E))

Probability is 0.22000000000000008


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. Consider two independent events X and Y. If the probability of occurrence of event X is 0.4 and the probability of occurrence of at least one event is 0.6. Find the probability of occurrence of event Y. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [8]:
# type your code here

prob_X = 0.4
prob_atleast_one = 0.6

prob_Y = (prob_atleast_one - prob_X) / (1 - prob_X)
print('Probability is {}'.format(prob_Y))

Probability is 0.33333333333333326


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. Consider a vending machine containing 6 boxes each of four different juice flavors (Mango, Apple, Guava and Kiwi). Find the probability that the juice box picked is either of Mango or Guava flavor.                 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [9]:
# type your code here

box = 6
flavor = 4
f_box = box * flavor

prob_M = box / f_box
prob_G = box / f_box

f_prob = prob_M + prob_G
print('Probability is {}'.format(f_prob))

Probability is 0.5


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>9. Consider a student appearing for 3 different exams either pass or fail in each exam. Let a marginal probability of a student passing the Economics, Mathematics and German exam be 0.67. The probability that the student fails in the Economics exam is 0.10 and the probability that the student fails in the German exam is 0.09. Find the probability that a student fails in the Mathematics exam.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [10]:
# type your code here

prob_P = 0.67
prob_F = 1 - prob_P

prob_of_failing_in_economics = 0.10
prob_of_failing_in_german = 0.09

prob_of_failing_in_maths = prob_F - (prob_of_failing_in_economics + prob_of_failing_in_german)
print('Probability is {}'.format(prob_of_failing_in_maths))

Probability is 0.13999999999999996


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>10. The probabilities that a machine will produce 2, 5, 7, 9 defective batteries in a day are 0.3, 0.22, 0.15, 0.1 respectively. Find the expected number of defective batteries produced in a day.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [11]:
# type your code here

no_of_defective_batteries = np.array([2, 5, 7, 9])
prob_of_defective_batteries = np.array([0.3, 0.22, 0.15, 0.1])

f_prob = np.sum(no_of_defective_batteries * prob_of_defective_batteries)

print('Expected number is {}'.format(f_prob))

Expected number is 3.65


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>11. In a pack of 52 cards, what is the probability of selecting a red card?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [12]:
# type your code here

total_cards = 52
red_cards = 26

prob = red_cards / total_cards
print('Probability is {}'.format(prob))

Probability is 0.5


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>12. From the experience, it is seen that 4% of the tyres produced by the machine are defective. Out of the 10 tyres produced, find the probability that at most 3 are defective.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [13]:
# type your code here

from scipy import stats
prob = stats.binom.pmf(0, 10, 0.04) + stats.binom.pmf(1, 10, 0.04) + stats.binom.pmf(2, 10, 0.04) + stats.binom.pmf(3, 10, 0.04)
print('Probability is {}'.format(prob))

Probability is 0.9995574006399994


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>13. The number of trucks crossing a bridge during the day follow a Poisson distribution with mean 18. What is the probability that on a randomly selected day 12 trucks would have crossed the bridge?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [14]:
# type your code here

from scipy import stats
prob = stats.poisson.pmf(12, 18)
print('Probability is {}'.format(prob))

Probability is 0.03678175293604137


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>14. A random variable follows uniform distribution in the interval 5 to 10. Obtain the mean and variance. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [15]:
# type your code here

lower_bound = 5
upper_bound = 10

mean = (upper_bound - lower_bound) / 2
print('Mean:', mean)

variance = (upper_bound - lower_bound) ** 2 / 12
print('Variance:', variance)

Mean: 2.5
Variance: 2.0833333333333335


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>15. Consider an random variable that follows normal distribution with mean 5 and variance 10 
obtain: <br>
1. P(X < 1)<br>
2. P(X > 5)<br>
3. P(3 < X < 7)
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [16]:
# type your code here

from scipy.stats import norm
mu = 5 
var =10
sd = np.sqrt(var)

print('P(X < 1): ',norm.cdf(1, mu, sd))
print('P(X > 5): ',1 - norm.cdf(5, mu, sd))
print('P(3 < X < 7): ',norm.cdf(6, 3, 2) - norm.cdf(4, 3, 2))

P(X < 1):  0.1029516053660342
P(X > 5):  0.5
P(3 < X < 7):  0.2417303374571288


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>16. Consider the data given below. Can we claim that the average number of fruits on a tree is 70? 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider number of fruits on a tree:

            number_fruits = [64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46]

In [17]:
# type your code here

number_fruits = pd.Series([64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46])

from scipy.stats import ttest_1samp
ttest_1samp(number_fruits, 70)

TtestResult(statistic=-1.36854291090804, pvalue=0.19270344894968697, df=14)

The p-value (0.1927) is greater than 0.05 (level of significance), hence we accept H0 and We can conclude that the average number of fruits on a tree is 70.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>17. The courier company has two offices in NYC. They are interested to know which office is busier. The number of parcels received for 15 days is given below. Can we claim that both offices are equally busy?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider the number of parcels in office A:
            office_A = [26, 56, 48, 56, 79, 67, 35, 46, 87, 53, 67, 89, 65, 42, 34]

            Consider the number of parcels in office B:
            office_B = [34, 45, 87, 57, 45, 42, 34, 45, 34, 67, 89, 43, 87, 95, 95]

In [18]:
# type your code here

parcels_office_A = pd.Series([64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46])
parcels_office_B = pd.Series([34, 45, 87, 57, 45, 42, 34, 45, 34, 67, 89, 43, 87, 95, 95])

from scipy.stats import ttest_ind
ttest_ind(parcels_office_A, parcels_office_B)

Ttest_indResult(statistic=0.6449141708280643, pvalue=0.5242321636740674)

The p-value (0.524) is more than 0.05 (level of significance), hence we accept H0 and We can conclude that the average number of parcels in office A is equal to the average number of parcels in office B.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>18. To improve the attitude of officers, they had to complete a training program. Can we claim that the training program was effective? 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

The data of 10 officers scores is given below:

<table>
    <tr>
        <th> Officer </th>
        <th> Attitude Before </th>
        <th> Attitude After </th>
    </tr>
     <tr>
        <td> 1 </td>
        <td> 3 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 2 </td>
        <td> 5 </td>
        <td> 5 </td>
    </tr>
    <tr>
        <td> 3 </td>
        <td> 2 </td>
        <td> 3 </td>
    </tr>
    <tr>
        <td> 4 </td>
        <td> 2 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 5 </td>
        <td> 4 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 6 </td>
        <td> 4 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 7 </td>
        <td> 3 </td>
        <td> 3 </td>
    </tr>
    <tr>
        <td> 8 </td>
        <td> 5 </td>
        <td> 2 </td>
    </tr>
    <tr>
        <td> 9 </td>
        <td> 3 </td>
        <td> 2 </td>
    </tr>
    <tr>
        <td> 10 </td>
        <td> 4 </td>
        <td> 5 </td>
    </tr>
</table>

In [19]:
# type your code here

score_before = pd.Series([3, 5, 2, 2, 4, 4, 3, 5, 3, 4])
score_after = pd.Series([4, 5, 3, 4, 4, 4, 3, 2, 2, 5])

from scipy.stats import ttest_rel
ttest_rel(score_before, score_after)

TtestResult(statistic=-0.2307692307692308, pvalue=0.8226545408271233, df=9)

The p-value is more than the level of significance, hence we accept H0 and We can conclude that there is no significant effect of the training.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>19. During an economic crisis, a researcher thought that the salaries of IT professionals are affected. Test whether the average monthly salary has gone below 65000 dollars.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Salaries of corporate professionals in thousand dollars:

            salary = [56, 48, 56, 79, 67, 55, 46, 87, 53, 67, 89, 65, 62, 45, 87, 57, 45, 
                      42, 45, 67, 89, 43, 87, 95, 92, 56, 67, 89, 54, 62, 78, 43, 76, 65,
                      56, 58, 98, 56, 78, 79, 45, 34, 35, 46, 47, 67, 37, 45, 51, 65, 65]

In [20]:
# type your code here

salaries = [56, 48, 56, 79, 67, 55, 46, 87, 53, 67, 89, 65, 62, 45, 87, 57, 45, 42, 45, 67, 89, 43, 87, 95, 92, 56, 67, 89, 54, 62, 78, 43, 76, 65, 56, 58, 98, 56, 78, 79, 45, 34, 35, 46, 47, 67, 37, 45, 51, 65, 65]
salaries = pd.Series(salaries)

from statsmodels.stats import weightstats as stests
stests.ztest(salaries, value = 65, alternative = "smaller")

(-1.1250934292594312, 0.1302747226990263)

The p-value is more than level of significance, hence we accept H0 and we can conclude that there is no change in the salaries of IT professionals.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>20. Consider the data given below. Can we conclude that the data is normally distributed?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider number of fruits on a tree:
            number_fruits = [64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46]

In [21]:
# type your code here

from scipy import stats

number_fruits = [64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46]
number_fruits = pd.Series(number_fruits)

stat, p = stats.shapiro(number_fruits)
print('The test statistic is', stat, 'and its p-value is', p)

The test statistic is 0.9354036450386047 and its p-value is 0.32807067036628723


The p-value is more than the level of significance, Therefore, We may conclude that the data is normally distributed.